lib/cacheWrapper.js

```diff
@@ -79,31 +115,33 @@ cacheWrapper._stash = function( deferred, options ) {
 * @param {Object} options - Including segment, key and value.
 * @returns {undefined} - nothing
 */
-cacheWrapper._queueRequest = function( deferred, options ) {
+cacheWrapper._queueRequest = function(deferred, options) {
   // add the request to the queue
-  queryQueue.push( [ deferred, options ] );
-
-  if ( !startingClient ) {
+  queryQueue.push([ deferred, options ]);
+  if (!startingClient) {
     startingClient = true;
-    cacheClient.start( function( error ) {
-      _.forEach( queryQueue, function( queueItem ) {
+    cacheClient.start(function(error) {
+      _.forEach(queryQueue, function(queueItem) {
         // As you can see above, we push the deferred and the options into the array
         var queuePromise = queueItem[0];
         var queueOptions = queueItem[1];
         // error, cache restart failed...
         // else the cache has now connected
-        if ( error ) {
-          queuePromise.reject( new Error( options.method + ' failed' ) );
-        } else if ( options.method === 'stash' ) {
-          cacheWrapper._stash( queuePromise, queueOptions );
+        if (error) {
+          queuePromise.reject(new Error(options.method + ' failed'));
+        } else if (options.method === 'stash') {
+
+          cacheWrapper._stash(queuePromise, queueOptions);
         } else {
-          cacheWrapper._retrieve( queuePromise, queueOptions );
+
+          cacheWrapper._retrieve(queuePromise, queueOptions);
         }
-      } );
+      });
+
       queryQueue = [];
```

Comment:
Do we need consider if there was an error starting the cache client, before reseting the queryQueue. If there was, then the queue item won't have been stashed/retrieved and will be lost forever.